### PR TITLE
PATCH - ensure node.js versions 4.0 to 4.4 use buffer fallback

### PR DESCRIPTION
**Pull request by evanshortiss**
Hi folks,

When using node v4.4.X I kept getting the following on startup:

```
TypeError: utf-8 is not a function
    at Function.from (native)
    at from (native)
    at Object.obfuscateNameUsingKey (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/util/hashes.js:31:22)
    at Config._fromServer (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/config.js:396:36)
    at Config.cb_forEach (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/config.js:359:10)
    at Array.forEach (native)
    at Config.onConnect (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/config.js:358:21)
    at Agent.reconfigure (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/agent.js:320:15)
    at cb_invoke (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/collector/api.js:181:15)
    at cb_nextTick (/Users/eshortis/workspaces/new-relic-integration/node_modules/newrelic/lib/collector/parse-response.js:113:7)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)
```

This is due to the `Buffer.from` implementation in node.js versions v4.0.x - v4.4.x differing from v4.5+. I've updated the shim you've provided for to account for this.

### CODE REVIEW

**Comment by evanshortiss** [comment]
Just realised I should add a unit test also. Will do ASAP.

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ language: node_js
 node_js:
   - "0.10"
   - "0.12"
+  - "4.4"
```
Comment (evanshortiss) [inline .travis.yml]: Added to help prevent regressions for this specific issue.

File: lib/util/hashes.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
 'use strict'
 
 var crypto = require('crypto')
+var process = require('./process')
```
Comment (evanshortiss) [inline lib/util/hashes.js]: `require('process')` is not available in node v0.10.x hence this shim in utils

File: test/unit/hashes.test.js [hunk 3]
```diff
@@ -17,3 +21,71 @@ describe('deobfuscation', function() {
...
   })
 })
+
+describe('Buffer.from fallback logic', function() {
+  var fromCalled = false
+
+  function getBufferStub(shouldFail) {
+    fromCalled = false
+
+    return xtend(Buffer, {
```
Comment (evanshortiss) [inline test/unit/hashes.test.js]: We make a clone of Buffer and stub the `from` function with a test implementation. I considered using `sinon` but wanted to add as few devDependencies as possible.

**Comment by lykkin** [comment]
hey, thanks @evanshortiss for the pr!  we're already tracking and have a fix for this issue we are planning on releasing in the very near future.

**Comment by evanshortiss** [comment]
@lykkin thanks! I assume I can close this so? 

Do you have a more specific timeline?

**Comment by lykkin** [comment]
@evanshortiss yeah, you are safe to close this.  we are shooting to release in the next couple of days.

**Comment by evanshortiss** [comment]
@lykkin appreciate it!

**Pull request closed without merging**